### PR TITLE
Add current limitations to readme and revise installation guide

### LINKED ISSUE #40: Document current limitations in `README.md`
@LittleBeannie has kindly offered to document the current limitations in `README.md` before we do the first release to CRAN.

The items roughly include:

- Might not work for cases where number of hypotheses grow.
- API might have breaking changes in future releases as they are being further reviewed and refactored.
- More validation is needed on logic.
- and more.

### PR DESCRIPTION
**Pull request by nanxstats**
Fixes #40

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,12 +14,31 @@ in the control arm.
...
+# install.packages("remotes")
 remotes::install_github("Merck/wpgsd")
 ```
 
+## Current limitations
+
+There are some limitations that are currently being addressed.
+Please use the package with caution in production environments.
+
+- The implementation may not be able to handle a larger number of hypotheses
```
Comment (LittleBeannie) [inline README.md:36]: Can we reword this sentence a little? 

"The current implementation may have limitations when handling more complex scenarios beyond what has been demonstrated."


Besides, can we add 1 more bullet that "More documentation is expected in the next release"?

**Review by LittleBeannie — COMMENTED** [review]
Thanks, Nan! I have 1 minor comment.

File: README.md (hunk 1 above)
Comment (nanxstats) [reply README.md]: Agreed! Updated.

**Review by LittleBeannie — APPROVED** [review]
Thanks, @nanxstats !

**Pull request merged**